firedrake/function.py

```diff
@@ -374,6 +374,16 @@ def interpolate(self, expression, subset=None, ad_block_tag=None):
         from firedrake import interpolation
         return interpolation.interpolate(expression, self, subset=subset, ad_block_tag=ad_block_tag)
 
+    def zero(self, subset=None):
+        """Set all values to zero.
+
+        :arg subset: :class:`pyop2.types.set.Subset` indicating the nodes to
+            zero. If ``None`` then the whole function is zeroed.
+        """
+        # Use assign here so we can reuse _ad_annotate_assign instead of needing
+        # to write an _ad_annotate_zero function
+        return self.assign(0, subset=subset)
+
```

Comment:
Is this annotated by `firedrake_adjoint`? Since it uses `assign` I presume the answer is yes?